spacelift/resource_drift_detection.go

```diff
@@ -0,0 +1,171 @@
+		DeleteContext: resourceDriftDetectionDelete,
+
+		Importer: &schema.ResourceImporter{StateContext: importIntegration},
+
+		Schema: map[string]*schema.Schema{
+			"reconcile": {
+				Type:        schema.TypeBool,
+				Description: "Whether a tracked run should be triggered when drift is detected.",
+				Optional:    true,
+			},
+			"stack_id": {
+				Type:        schema.TypeString,
+				Description: "ID of the stack for which to set up drift detection",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"schedule": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				MinItems:    1,
+				Description: "List of cron schedule expressions based on which drift detection should be triggered.",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourceDriftDetectionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var mutation struct {
+		CreateDriftDetectionIntegration struct {
+			Reconcile bool     `graphql:"reconcile"`
+			Schedule  []string `graphql:"schedule"`
+		} `graphql:"stackIntegrationDriftDetectionCreate(stack: $stack, input: $input)"`
+	}
+
+	var scheduleExpressions []graphql.String
+	for _, expr := range d.Get("schedule").([]interface{}) {
+		scheduleExpressions = append(scheduleExpressions, graphql.String(expr.(string)))
+	}
+
+	stackID := d.Get("stack_id").(string)
+
+	variables := map[string]interface{}{
+		"stack": toID(stackID),
+		"input": structs.DriftDetectionIntegrationInput{
+			Reconcile: graphql.Boolean(d.Get("reconcile").(bool)),
+			Schedule:  scheduleExpressions,
+		},
+	}
+
+	if err := meta.(*internal.Client).Mutate(ctx, "DriftDetectionUpdate", &mutation, variables); err != nil {
+		return diag.Errorf("could not create drift detection integration for the stack: %v", err)
+	}
+
+	if d.Id() == "" {
+		d.SetId(stackID)
+	}
+
+	return resourceDriftDetectionRead(ctx, d, meta)
+}
+
+func resourceDriftDetectionUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var mutation struct {
+		CreateDriftDetectionIntegration struct {
+			Reconcile bool     `graphql:"reconcile"`
+			Schedule  []string `graphql:"schedule"`
+		} `graphql:"stackIntegrationDriftDetectionUpdate(stack: $stack, input: $input)"`
+	}
+
+	var scheduleExpressions []graphql.String
+	for _, expr := range d.Get("schedule").([]interface{}) {
+		scheduleExpressions = append(scheduleExpressions, graphql.String(expr.(string)))
+	}
+
+	stackID := d.Get("stack_id").(string)
+
+	variables := map[string]interface{}{
+		"stack": toID(stackID),
+		"input": structs.DriftDetectionIntegrationInput{
+			Reconcile: graphql.Boolean(d.Get("reconcile").(bool)),
+			Schedule:  scheduleExpressions,
+		},
+	}
+
+	if err := meta.(*internal.Client).Mutate(ctx, "DriftDetectionCreate", &mutation, variables); err != nil {
+		return diag.Errorf("could not update drift detection integration for the stack: %v", err)
+	}
+
+	if d.Id() == "" {
```

Comment:
You're not supposed to do that in the `update`. The assumption is that the ID is set (and immutable) for a resource that's being updated.